lib/system/web/src/keyboard_engine.rs

```diff
@@ -0,0 +1,170 @@
+use std::collections::{HashMap, BTreeSet};
+use std::cell::RefCell;
+use std::rc::Rc;
+use wasm_bindgen::JsCast;
+use wasm_bindgen::prelude::*;
+use itertools::Itertools;
+
+enum Action {
+    KeyUp,
+    KeyDown
+}
```

Comment:
The name `Action` is misleading. `KeyUp` or `KeyDown` are not actions - they are more like the current state of a key. Moreover, it's always preferred to use short names, what about some modules and names like `key::state::Up` instead?

Reply:
I decided not to create separated function, hence I've removed the seciont